Mathlib/Tactic/MoveAdd.lean

```diff
@@ -0,0 +1,379 @@
+/-
+Copyright (c) 2023 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino, Damiano Testa
+-/
+import Mathlib.Algebra.Group.Basic
+
+/-!
+
+#  `move_add` a tactic for moving summands in expressions
+
+The tactic `move_add` rearranges summands in expressions.
+It does so by receiving as input a list of "summand atoms", inducing the desired permutation.
+See the doc-string for `move_oper` for details of how the permutation is determined by the input.
+For the technical description, look at `Mathlib.MoveAdd.weight` and `Mathlib.MoveAdd.reorderUsing`.
+
+`move_add` is the specialization of a more general `move_oper` tactic that takes a binary,
+associative, commutative operation and a list of "operand atoms" and rearranges the operation.
+
+## Extension notes
+
+To work with a general associative, commutative binary operation, `move_oper`
+needs to have inbuilt the lemmas asserting the analogues of
+`add_comm, add_assoc, add_left_comm` for the new operation.
+Currently, `move_oper` supports `HAdd.hAdd`, `HAdd.hAdd`, `and`, `or`, `max`, `min`.
+
+These lemmas should be added to `Mathlib.MoveAdd.move_oper_simpCtx`.
+
+## Implementation notes
+
+The tactic takes the target, replaces the maximal subexpressions whose head symbol is the given
+operation and replaces them by their reordered versions.
+Once that is done, it tries to replace the initial goal with the permuted one by using `simp`.
+
+Currently, no attempt is made at guiding `simp` by doing a `congr`-like destruction of the goal.
+I, DT, already have code for this improvement, but am leaving it for a later PR.
```

Comment:
```suggestion
I, DT, already have code for this improvement, but am leaving it for a later PR.
```
I think to the extent we put names in implementation notes we use 3rd person. :-)